core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/uri/DefaultResourceURIPathResolver.java

```diff
@@ -75,13 +77,18 @@ private String resolve(ServiceProvider context, CodeSystemURI uriToResolve, Code
 					.findFirst()
 					.map(CodeSystemVersion::getPath)
 					.orElseGet(() -> {
-						if (uriToResolve.isLatest()) {
+						if (uriToResolve.isLatest() || !allowBranches) {
```

Comment:
I don't understand the reason behind this addition here? The unit tests are not covering this boolean property either. Is this used for making sure the upgrade code system version does exist? If yes then I think that must be handled somewhere else, probably in the CodeSystemUpgradeRequest